src/window-service.js

```diff
@@ -196,20 +196,53 @@
             mainWindow.addEventListener('closed', closedEvent);
         }
 
-        snapToScreenBounds(targetWindow) {
+        getTargetMonitor(x, y, callback) {
             fin.desktop.System.getMonitorInfo((info) => {
-                targetWindow.getBounds((bounds) => {
+                var monitors = info.nonPrimaryMonitors.concat(info.primaryMonitor);
+                let closestMonitor = monitors[0];
+                let closestDistance = Number.MAX_VALUE;
+
+                for (var monitor of monitors) {
+
+                    // If the window's top-left is within the monitor's bounds, use that + stop
+                    if (x >= monitor.monitorRect.left && x <= monitor.monitorRect.right &&
+                            y >= monitor.monitorRect.top && y <= monitor.monitorRect.bottom) {
+
+                        callback(monitor);
+                        return;
+
+                    } else {
+
+                        // Otherwise, keep track of the closest, and if the window is not
+                        // within any monitor bounds, use the closest.
+                        var midX = monitor.monitorRect.left + monitor.monitorRect.right / 2;
+                        var midY = monitor.monitorRect.top + monitor.monitorRect.bottom / 2;
+                        var distance = Math.pow(midX - x, 2) + Math.pow(midY - y, 2);
+                        if (distance < closestDistance) {
+                            closestDistance = distance;
+                            closestMonitor = monitor;
+                        }
+
+                    }
+                }
+                callback(closestMonitor);
```

Comment:
Maybe `monitor.monitorRect` should be a variable? And is there anything in the geometryService (or that should be added to it) that could help out here?